docs/documentation/code-format-lint.md

```diff
@@ -0,0 +1,98 @@
+# Linting and Code Formatting
```

Comment:
maybe add some discussion on hatch fmt (which only runs ruff). We want to run codespell and markdownlint and others potentially. What is the best way to do this?

https://hatch.pypa.io/latest/config/internal/static-analysis/

Reply:
I looked into using `hatch fmt` with Jupyter projects but decided against it, since without using https://hatch.pypa.io/latest/config/internal/static-analysis/#persistent-config the rules are opaque and not visible to editors, and I didn't want to commit such a large file to CI.

Reply:
ok this makes sense. so we should for now, not use that. 

how does jupyter handle linting or where do you all land?